tikv-client/src/main/java/com/pingcap/tikv/key/IndexKey.java

```diff
@@ -42,54 +43,51 @@ private IndexKey(long tableId, long indexId, Key[] dataKeys) {
   }
 
   public static class EncodeIndexDataResult {
-    public EncodeIndexDataResult(Key[] keys, boolean appendHandle) {
+    public EncodeIndexDataResult(byte[] keys, boolean appendHandle) {
       this.keys = keys;
       this.appendHandle = appendHandle;
     }
 
-    public Key[] keys;
+    public byte[] keys;
     public boolean appendHandle;
   }
 
   public static IndexKey toIndexKey(long tableId, long indexId, Key... dataKeys) {
     return new IndexKey(tableId, indexId, dataKeys);
   }
 
-  public static EncodeIndexDataResult encodeIndexDataValues(
-      Row row,
-      List<TiIndexColumn> indexColumns,
-      Handle handle,
-      boolean appendHandleIfContainsNull,
-      TiTableInfo tableInfo) {
+  public static EncodeIndexDataResult genIndexKey(
+      long physicalID, Row row, TiIndexInfo indexInfo, Handle handle, TiTableInfo tableInfo) {
     // when appendHandleIfContainsNull is true, append handle column if any of the index column is
     // NULL
-    boolean appendHandle = false;
```

Comment:
why do you want to change appendHandle to distinct?
and if you changed the name, you need to change the place where you use it. such as https://github.com/pingcap/tispark/blob/6b6ac15859515145c16cfefedb5d2ac5e7ca8f01/core/src/main/scala/com/pingcap/tispark/utils/WriteUtil.scala#L238

Reply:
> why do you want to change appendHandle to distinct? and if you changed the name, you need to change the place where you use it. such as
> 
> https://github.com/pingcap/tispark/blob/6b6ac15859515145c16cfefedb5d2ac5e7ca8f01/core/src/main/scala/com/pingcap/tispark/utils/WriteUtil.scala#L238

I renamed here